src/Runner.Worker/ActionRunner.cs

```diff
@@ -89,7 +91,18 @@ public async Task RunAsync()
                 string.Equals(localAction.RepositoryType, Pipelines.PipelineConstants.SelfAlias, StringComparison.OrdinalIgnoreCase))
             {
                 var actionManager = HostContext.GetService<IActionManager>();
-                var prepareResult = await actionManager.PrepareActionsAsync(ExecutionContext, compositeHandlerData.Steps, ExecutionContext.Id);
+                PrepareResult prepareResult = default(PrepareResult);
+                try
+                {
+                    prepareResult = await actionManager.PrepareActionsAsync(ExecutionContext, compositeHandlerData.Steps, ExecutionContext.Id);
+                }
+                catch (Exception ex)
+                {
+                    if (ex is WebApi.FailedToResolveActionDownloadInfoException)
+                    {
+                        throw new WebApi.FailedToResolveActionDownloadInfoException("Failed to resolve action download info", ex);
+                    }
+                }
```

Comment:
```suggestion
                catch (WebApi.FailedToResolveActionDownloadInfoException ex)
                {
                    throw new WebApi.FailedToResolveActionDownloadInfoException("Failed to resolve action download info", ex);
                }
```

Otherwise, we're silently swallowing all other types of exception.

Reply:
+1.

I think we also want to include in this block the areas that log this as an infrastructure error from [here](https://github.com/actions/runner/blob/9958fc0374820841ec0c131fa1259a442593fb1b/src/Runner.Worker/JobExtension.cs#LL436-L440C27)